src/boards/rainbow_esp.cpp

```diff
@@ -484,15 +485,15 @@ void BrokeStudioFirmware::processBufferedMessage() {
 			break;
 		case n2e_cmds_t::FILE_WRITE:
 			UDBG("RAINBOW BrokeStudioFirmware received message FILE_WRITE\n");
-			if (message_size >= 3 && this->rx_buffer[2] == message_size - 2 && this->working_file != NO_WORKING_FILE) {
-				this->writeFile(this->working_path, this->working_file, this->file_offset, this->rx_buffer.begin() + 3, this->rx_buffer.begin() + message_size + 1);
-				this->file_offset += message_size - 2;
+			if (message_size >= 3 && this->working_file != NO_WORKING_FILE) {
+				this->writeFile(this->working_path, this->working_file, this->file_offset, this->rx_buffer.begin() + 2, this->rx_buffer.begin() + 2 + message_size - 1);
+				this->file_offset += message_size - 1;
 			}
 			break;
 		case n2e_cmds_t::FILE_APPEND:
 			UDBG("RAINBOW BrokeStudioFirmware received message FILE_APPEND\n");
-			if (message_size >= 3 && this->rx_buffer[2] == message_size - 2 && this->working_file != NO_WORKING_FILE) {
-				this->writeFile(this->working_path, this->working_file, this->files[working_path][working_file].size(), this->rx_buffer.begin() + 3, this->rx_buffer.begin() + message_size + 1);
+			if (message_size >= 3 && this->working_file != NO_WORKING_FILE) {
+				this->writeFile(this->working_path, this->working_file, this->files[working_path][working_file].size(), this->rx_buffer.begin() + 2, this->rx_buffer.begin() + 2 + message_size - 1);
```

Comment:
Not sure why you changed the end iterator computation the new version seems logically equivalent to the former one, but with some undefined behaviour.


If the message is the last in the buffer, `this->rx_buffer.begin() + 2 + message_size - 1` sould resolve to `rx_buffer.end()`, but the intermediate value `begin() + 2 + message_size` is after `end()` which is invalid.

You could write it `this->rx_buffer.begin() + 2 + (message_size - 1)` , but once again I don't see the point of changin it in the first place.

Reply:
Now that you point it I can see it !
I had a weird bug while trying to write data to a file. I tried different things, but not sure in the end if it was my ASM code fault, or ESP code... 
Anyway, I reverted back the code.